core/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsAnswer.java

```diff
@@ -0,0 +1,49 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.diagnostics;
+
+import com.cloud.agent.api.Answer;
+import com.cloud.utils.exception.CloudRuntimeException;
+import org.apache.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DiagnosticsAnswer extends Answer {
+    public static final Logger LOGGER = Logger.getLogger(DiagnosticsAnswer.class);
+
+    public DiagnosticsAnswer(DiagnosticsCommand cmd, boolean result, String details) {
+        super(cmd, result, details);
+    }
+
+    public Map<String, String> getExecutionDetails() {
+        final Map<String, String> executionDetailsMap = new HashMap<>();
+        if (result == true){
+            final String[] parseDetails = details.split("}");
+            if (parseDetails.length >= 3 ){
+                executionDetailsMap.put("STDOUT", parseDetails[0].trim());
```

Comment:
Refactor the use of Map keys to some interface/class, instead of hardcoding for STDOUT, STDERR and EXITCODE. The parsing may be improved here (see comment in the vr diagnostics py file).